pkg/controller/deployment/deployment_controller.go

```diff
@@ -424,17 +434,60 @@ func (dc *DeploymentController) syncDeployment(key string) error {
 	return fmt.Errorf("unexpected deployment strategy type: %s", d.Spec.Strategy.Type)
 }
 
-func (dc *DeploymentController) syncRecreateDeployment(deployment extensions.Deployment) error {
-	newRC, err := dc.getNewRC(deployment)
+// Rolling back to a revision; no-op if the toRevision is deployment's current revision
+func (dc *DeploymentController) rollback(deployment extensions.Deployment, toRevision *int) error {
+	if *toRevision < 0 {
+		dc.eventRecorder.Eventf(&deployment, api.EventTypeWarning, "RollbackDeployment", "Invalid revision to rollback to. Gives up rollback.")
+		// Gives up rollback
+		_, err := dc.cleanUpRollbackTo(&deployment)
+		return err
+	}
+	newRC, allOldRCs, err := dc.getNewRCAndAllOldRCs(deployment)
 	if err != nil {
+		dc.eventRecorder.Eventf(&deployment, api.EventTypeWarning, "RollbackDeployment", "Failed to rollback deployment %s to revision %d: %v", deployment.Name, *toRevision, err)
 		return err
 	}
+	allRCs := append(allOldRCs, newRC)
+	// If rollback revision is 0, rollback to the last revision
+	if *toRevision == 0 {
+		*toRevision = secondMaxRevision(allRCs)
+	}
+	for _, rc := range allRCs {
+		v, err := revision(rc)
+		if err != nil {
+			glog.V(4).Infof("Unable to extract revision from deployment's RC %+v: %v", rc, err)
+			continue
+		}
+		glog.V(4).Infof("Found rc with revision %d: %+v", v, rc)
+		if v == *toRevision {
+			glog.V(4).Infof("Found a match!")
```

Comment:
Squash this with the previous log, and make that more meaningful eg. "Desired revision: %d, found rc %q with revision %d"